sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -345,6 +345,14 @@ private[spark] object SQLConf {
     defaultValue = Some(true),
     doc = "Enables using the custom ParquetUnsafeRowRecordReader.")
 
+  // Note: this can not be enabled all the time because the reader will not be returning UnsafeRows.
+  // Doing so is very expensive and we should remove this requirement instead of fixing it here.
+  // Initial testing seems to indicate only sort requires this.
```

Comment:
Right, all the operators output UnsafeRow, some operators may depends on some properties of UnsafeRow: 

1). copy() returns UnsafeRow

2). getStruct() return UnsafeRow, getArray() return UnsafeArrayData, getMap() returns UnsafeMap

3). hashCode() is murmur3 on bytes of UnsafeRow

4). compareTo() will compare the row as bytes

For example,  the in-memory cache requires 2), except requires 4)

@rxin Do you have more comment on this?